File: src/main/java/net/swade/riot/api/objects/Summoner.java
package net.swade.riot.api.objects;

import lombok.Getter;

public record Summoner(@Getter String accountId, @Getter int profileIconId, @Getter String revisionDate,
                       @Getter String name, @Getter String id, @Getter String puuid, @Getter String summonerLevel) {
}

File: src/main/kotlin/objects/lol/Summoner.kt
package net.guneyilmaz0.riotapi.objects.lol

@Suppress("unused")
data class Summoner(
    val accountId: String,
    val profileIconId: Int,
    val revisionDate: Long,
    val name: String,
    val id: String,
    val puuid: String,
    val summonerLevel: Int
)